radis/test/tools/test_database.py

```diff
@@ -115,17 +115,67 @@ def test_plot_spec(plot=True, close_plots=True, verbose=True, *args, **kwargs):
         plot_spec(getTestFile("N2C_specair_380nm.spec"))
 
 
-def _run_testcases(plot=True, close_plots=False, verbose=True, *args, **kwargs):
+def test_save_compressed2(verbose=True, *args, **kwargs):
+    "Check if saving a spectrum with compress = 2 does not change something."
+    import shutil
 
-    test_speclist(*args, **kwargs)
-    test_database_functions(
-        plot=plot, close_plots=close_plots, verbose=verbose, *args, **kwargs
-    )
-    test_plot_spec(plot=plot, close_plots=close_plots, verbose=verbose, *args, **kwargs)
+    from radis.test.utils import setup_test_line_databases
+    from radis import calc_spectrum, SpecDatabase
 
-    return True
+    try:
+        # we want to make sure this folder is not generated from before
+        shutil.rmtree(dirname(getTestFile(".")) + "/newDb/")
+    except:
+        pass
+    
+    try:
+        #get the spectrum
+        setup_test_line_databases()
+        s = calc_spectrum(
+            2000,
+            2300,  # cm-1
+            molecule="CO",
+            isotope="1,2,3",
+            pressure=1.01325,  # bar
+            Tgas=700,  # K
+            mole_fraction=0.1,
+            path_length=1,  # cm
+            verbose=False,
+            wstep=0.01,
+            medium="vacuum",
+            databank="HITRAN-CO-TEST",
+        )
 
+        # load in one databse
+        db = SpecDatabase(dirname(getTestFile(".")) + "/newDb/")
+        db.add(s, compress=2, if_exists_then="error")
+    
+        # simulate an experimentalist who come later and load the spectrum
+        db2 = SpecDatabase(dirname(getTestFile(".")) + "/newDb/")
+        s_bis = db2.get_unique(Tgas=700)
+        
+        # we clean the folders for next times
+        shutil.rmtree(dirname(getTestFile(".")) + "/newDb/")
+    
+    except:
```

Comment:
```suggestion
    finally:
```

Very different from `except`

Also, you should start the `try` on line 149: we shouldn't try to clean the folders before they were created